### ISSUE TITLE
feat: Material UI TimePicker

### ISSUE THREAD

**Issue by Lindsor** [ionic-team/ionic-framework#27842]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

Ionic currently only implements ion-datetime as 1 element which either displays as a calendar for dates or a spinner for time.
On Android google material-ui TimePicker has a very specific design, which personally i believe is one of the best time pickers out there, since Ionic implements Android/IOs native styles we should have the angular-ui TimePicker as well.

I believe Ionic is already looking at Material UI 3 so i have included below both MUI2 and MUI3 links.

Material UI TimePicker docs m3:
https://m3.material.io/components/time-pickers/overview
<img width="1044" alt="image" src="https://github.com/ionic-team/ionic-framework/assets/3959304/379c4140-41f8-4047-9ca4-bfde662e9729">

Material UI TimePicker docs m2:
https://m2.material.io/components/time-pickers
<img width="799" alt="image" src="https://github.com/ionic-team/ionic-framework/assets/3959304/6f75ebe2-1f99-458b-930e-a94a45b8a19d">

NOTE: This was originally brought up by Yohandah in https://github.com/ionic-team/ionic-framework/issues/22152 however it was closed with no actual implementation.

### Describe the Use Case

Ionic implementa IOs / Android native styles this should help complete that.

**Comment by liamdebeasi** [comment]
Thanks for the feature request! I discussed this with the team, and we have no plans to implement this at the moment. We considered it back when we re-designed `ion-datetime` for Ionic 6, but we opted not to implement it then either. 

Our current stance is the value added by this feature already exists in `ion-datetime`. More specifically: The proposed feature calls for swipe gestures to select the time as well as an option for time entry via a keyboard. Both of those features exist when selecting the time with the wheel picker in `ion-datetime`.

With datetime we prioritized feature and interface consistency across iOS and MD. While this can result in us diverging from the spec, it ensures that users are able to do the same date/time selection actions across iOS and MD.
